src/dispatch/case/views.py

```diff
@@ -167,6 +172,46 @@ def update_case(
     return case
 
 
+@router.put(
+    "/{case_id}/escalate",
+    response_model=IncidentRead,
+    summary="Escalates an existing case.",
+    # dependencies=[Depends(PermissionsDependency([CaseEditPermission]))],
+)
+def escalate_case(
+    *,
+    db_session: Session = Depends(get_db),
+    current_case: Case = Depends(get_current_case),
+    organization: OrganizationSlug,
+    incident_in: IncidentCreate,
+    current_user: DispatchUser = Depends(auth_service.get_current_user),
+    background_tasks: BackgroundTasks,
+):
+    """Escalates an existing case."""
+    # current user is better than assignee (although likely the same)
+    if not incident_in.reporter:
+        incident_in.reporter = ParticipantUpdate(
+            individual=IndividualContactRead(email=current_user.email)
+        )
+
+    # allow for default values
+    if not incident_in.incident_type:
+        incident_in.incident_type = {"name": current_case.case_type.incident_type.name}
```

Comment:
We're assuming that the case type is mapped to an incident type here, which may not always be the case. We should check if the mapping exists, and if not, default to the default incident type on that project?